rskj-core/src/integrationTest/java/pte/PteIntegrationTest.java

```diff
@@ -0,0 +1,270 @@
+                baseJavaCmd,
+                buildLibsPath,
+                jarName,
+                strBaseArgs);
+
+        // When
+
+        CommandLineFixture.runCommand(
+                cmd,
+                1,
+                TimeUnit.MINUTES,
+                proc -> {
+                    try {
+
+                        // Send bulk transactions
+
+                        Response txResponse = sendBulkTransactions(
+                                accounts.get(0), accounts.get(1), accounts.get(2), accounts.get(3),
+                                accounts.get(4), accounts.get(5), accounts.get(6), accounts.get(7));
+
+                        txResponseMap.put("bulkTransactionsResponse", txResponse);
+
+                        // Await for n blocks to be mined and return them
+
+                        Future<Map<Integer, String>> future = getBlocksAsync();
+
+                        try {
+
+                            blocksResponseMap.put("asyncBlocksResult", future.get());
+
+                        } catch (ExecutionException | InterruptedException e) {
+                            e.printStackTrace();
+                            Assertions.fail(e);
+                        }
+
+                    } catch (IOException e) {
+                        Assertions.fail(e);
+                    }
+                }
+        );
+
+        // Then
+
+        Assertions.assertEquals(200, txResponseMap.get("bulkTransactionsResponse").code());
+
+        Map<Integer, String> blocksResult = blocksResponseMap.get("asyncBlocksResult");
+        Assertions.assertEquals(MAX_BLOCKS_TO_GET, blocksResult.size());
+
+        boolean pteFound = false;
+        int i = blocksResult.size(); // Start from the last element (optimization)
+        while (!pteFound && i >= 0) {
+            i -= 1;
+
+            JsonNode blockResponse = objectMapper.readTree(blocksResult.get(i));
+            JsonNode result = blockResponse.get(0).get("result");
+
+            if (!result.isNull()) {
+                JsonNode pteEdges = result.get("rskPteEdges");
+                if (pteEdges.isArray() && pteEdges.size() > 0) {
+                    Assertions.assertTrue(result.get("transactions").isArray());
+                    Assertions.assertTrue(result.get("transactions").size() > 1);
+                    pteFound = true;
+                }
+            }
+        }
+
+        Assertions.assertTrue(pteFound);
+
+    }
+
+    private Response getBlockByNumber(String number) throws IOException {
+        String content = "[{\n" +
+                "    \"method\": \"eth_getBlockByNumber\",\n" +
+                "    \"params\": [\n" +
+                "        \"" + number + "\",\n" +
+                "        true\n" +
+                "    ],\n" +
+                "    \"id\": 1,\n" +
+                "    \"jsonrpc\": \"2.0\"\n" +
+                "}]";
+
+        return OkHttpClientTestFixture.sendJsonRpcMessage(content, RPC_PORT);
+    }
```

Comment:
suggestion:

move this method to the class `OkHttpClientTestFixture`,  create a constant like `GET_BEST_BLOCK_CONTENT` with the body and call directly the method here. This way, we will increase the number of RPC method present in the class `OkHttpClientTestFixture` and re-use them in future tests. As example, you can take a look on the method `sendJsonRpcGetBestBlockMessage` [here](https://github.com/rsksmart/rskj/blob/master/rskj-core/src/integrationTest/java/co/rsk/util/OkHttpClientTestFixture.java#L96).

Reply:
Will do! really good idea!